src/utils.jl

```diff
@@ -10,3 +10,21 @@ end
 
 # Extract the length units used by the CRS of a Point
 _units(pt::Meshes.Point{M, CRS}) where {M, CRS} = first(CoordRefSystems.units(CRS))
+
+# Meshes.Vec -> (units, ::CliffordNumber.KVector)
+function _kvector(v::Meshes.Vec{Dim, T}) where {Dim, T}
+    units = Unitful.unit(T)
+    ucoords = Iterators.map(x -> Unitful.ustrip(units, x), v.coords)
+    kvec = CliffordNumbers.KVector{1, VGA(Dim)}(ucoords...)
+    return (units, kvec)
```

Comment:
I’ll take another look at this later today. Looking at the source for `Meshes.Vec` confirms that the vector components have units that are the same by definition, so you’re right that this can be simplified a bit.

Reply:
It turned out that the `_units` function actually already did what we were looking for, just specifically for a `Point`. I abstracted it to work for all geometry types and used that here.